examples/react/virtualized-rows/src/windowHeightTable.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react'
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useWindowVirtualizer } from '@tanstack/react-virtual'
+
+import { Person } from './makeData'
+
+type WindowHeightTableProps = {
+  // The data to render
+  data: any
+  // The columns to render
+  columns: ColumnDef<Person>[]
+}
+
+/**
+ * Renders full window height virtualised table
+ */
+export function WindowHeightTable({ data, columns }: WindowHeightTableProps) {
+  const [sorting, setSorting] = React.useState<SortingState>([])
+
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true,
+  })
+
+  const tableContainerRef = React.useRef<HTMLDivElement>(null)
+
+  // Get offset from top of window to ensure space is left for any
+  // html elements above the table
+  const tableOffsetRef = React.useRef(0)
+  React.useLayoutEffect(() => {
+    tableOffsetRef.current = tableContainerRef.current?.offsetTop ?? 0
+  }, [])
+
+  const { rows } = table.getRowModel()
+  const rowVirtualizer = useWindowVirtualizer({
+    count: rows.length,
+    scrollMargin: tableOffsetRef.current,
+    estimateSize: () => 54,
+    overscan: 10,
+  })
+
+  const virtualRows = rowVirtualizer.getVirtualItems()
+
+  // We need to add a padding row at top that grows as we scroll down to ensure
+  // the table scrolls down with the page
+  const paddingTop =
+    virtualRows.length > 0
+      ? virtualRows?.[0]?.start
+        ? virtualRows?.[0]?.start - tableOffsetRef.current
+        : 0 || 0
+      : 0
```

Comment:
We can use almost same logic as in, just here we need to remove also scrollMargin as we use it 

```suggestion
 const [paddingTop, paddingBottom] =
    virtualRows.length > 0
      ? [
          Math.max(0, virtualRows[0].start - rowVirtualizer.options.scrollMargin),
          Math.max(0, rowVirtualizer.getTotalSize() - virtualRows[virtualRows.length - 1].end),
        ]
      : [0, 0]
```